protocols/gossipsub/src/config.rs

```diff
@@ -390,6 +477,20 @@ impl Config {
     pub fn idontwant_on_publish(&self) -> bool {
         self.idontwant_on_publish
     }
+
+    /// Configuration for topics including mesh and max message transmit sizes.
+    pub fn topic_configuration(&self) -> &TopicConfigs {
+        &self.topic_configuration
+    }
+
+    /// The topic configuration sets mesh parameters for a given topic. See [`TopicMeshConfig`]
+    /// for defaults.
+    pub fn set_topic_config(&mut self, topic: TopicHash, config: TopicMeshConfig) -> &mut Self {
```

Comment:
Is there a reason why we should set the topic in the config itself?

Reply:
Set topic will be removed from Config as it was only meant to be on ConfigBuilder. It's just a duplicate I placed there while refactoring